src/promptflow/promptflow/executor/_async_nodes_scheduler.py

```diff
@@ -114,7 +118,8 @@ def _create_node_task(
         f = self._tools_manager.get_tool(node.name)
         kwargs = dag_manager.get_node_valid_inputs(node, f)
         if inspect.iscoroutinefunction(f):
-            task = context.invoke_tool_async(node, f, kwargs)
+            coroutine = context.invoke_tool_async(node, f, kwargs)
+            task = self.run_task_with_semaphore(coroutine)
```

Comment:
Seems only async task is under the concurrency limit. Sync/async tasks should share the same limit.

Reply:
Added run_task_with_semaphore to sync tasks.